src/rayleigh_quotient.jl

```diff
@@ -83,70 +116,113 @@ end
 
 #https://www.cis.upenn.edu/~jshi/papers/supplement_nips2006.pdf
 function solve(prob::ConstrainedRayleighQuotientProblem, alg::RQ_EIG)
-    b = prob.b
+    b = _get_b(prob) # fetch b matrix/vector if prob.b is span/vector
     C = prob.C
-    Kb = I - b * b' / dot(b, b)
-    nlargest = partialsortperm(eachrow(b), 1:size(b, 2); by=norm, rev=true)
     N = size(b, 1)
-    inds = map(!in(nlargest), 1:N)
-    J = I(N)[inds, :]
-    augC = J * Kb * C
+    Kb = I - b * b' / dot(b, b)
+    _, inds_keep = _inds_to_remove_and_keep(b, N)
+    augC = (Kb * C)[inds_keep, :] # instead of J * Kb * C where J = I(N)[inds_keep, :]
+    return _solve_homo_prob(augC, prob, alg)
+end
+
+function solve(prob::ConstrainedRayleighQuotientProblem{Q,Cmat,<:AbstractVector}, alg::RQ_SPARSE) where {Q,Cmat}
+    b = _get_b(prob)
+    C = prob.C
+    inds_remove, inds_keep = _inds_to_remove_and_keep(b, length(b))
+    bk = first(b[inds_remove]) # only one element, no support for b Span
+    Ck = C[inds_remove, :]
+    augC = (C - (1/bk) * b * Ck)[inds_keep, :] # replace mult by J with slicing, see above
+    return _solve_homo_prob(augC, prob, alg)
+end
+
+function _inds_to_remove_and_keep(b, N)
+    inds_remove = partialsortperm(eachrow(b), 1:size(b, 2); by=norm, rev=true)
+    inds_keep = map(!in(inds_remove), 1:N)
+    return inds_remove, inds_keep
+end
+
+function _solve_homo_prob(augC, prob::ConstrainedRayleighQuotientProblem, alg)
+    b = _get_b(prob)
+    C = prob.C
     homo_prob = ConstrainedRayleighQuotientProblem(prob.Q, augC, zero(b))
     sol = solve(homo_prob, RQ_HOMO(alg.eps))
-    z = dot(b, C, sol) / dot(b, b)
-    return sol / z
+    t = dot(b, C, sol) / dot(b, b)
+    return sol / t
+end
+
+function _solve_homo_prob(augC, prob::ConstrainedRayleighQuotientProblem{Q,Cmat,<:Span} , alg) where {Q,Cmat}
+    b = _get_b(prob)[:, 1] # for b Span, normalization is not specified
+    homo_prob = ConstrainedRayleighQuotientProblem(prob.Q, augC, zero(b))
+    return solve(homo_prob, RQ_HOMO(alg.eps)) # returns a view, is that ok?
 end
 
 function solve(prob::ConstrainedRayleighQuotientProblem, alg::RQ_HOMO)
     @assert iszero(prob.b) "b must be zero"
     C = prob.C
     P = I - C' * (factorize(Hermitian(C * C')) \ C)
```

Comment:
I guess P will be a dense matrix because we do `F\C`, where F is the factorization. We may want to define it only through its action on a vector instead. Then we only need to do `F\(C*v)`.